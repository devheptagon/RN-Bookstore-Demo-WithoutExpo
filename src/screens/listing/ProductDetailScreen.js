import React from 'react'
import PropTypes from 'prop-types'
import Focusable from '../hoc/Focusable'
import ProductInfo from '../listing/components/ProductInfo'

export class ProductDetailScreen extends React.Component {
  static navigationOptions = {
    header: null
  };

  render() {
    const { navigation } = this.props;
    const data = navigation.getParam('data', {});
    return (
      <Focusable>
        <ProductInfo dataItem={data} isDetailed={true} />
      </Focusable>
    );
  }
}

ProductDetailScreen.propTypes = {  
  navigation: PropTypes.object  
}

export default ProductDetailScreen