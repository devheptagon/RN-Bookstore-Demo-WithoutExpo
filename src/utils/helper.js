import React from "react";
import { FormattedCurrency } from "react-native-globalize";
import { createStackNavigator, createAppContainer } from "react-navigation";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import AppConfig from "../config/appConfig";

export const formatMoney = amount => {
  return <FormattedCurrency value={amount} />;
};

export const getRandomInt = (min, max) => {
  const rnd = Math.floor(Math.random() * Math.floor(max));
  return rnd < min ? min : rnd;
};

export const buildQueryUrl = (keyword, page) => {
  const { apiBaseUrl, pageSize } = AppConfig;
  return `${apiBaseUrl}?q=intitle:${keyword}&startIndex=${(page - 1) *
    pageSize}&maxResults=${pageSize}`;
};

export const buildDTOList = data => {
  let result = [];
  data.map(d => {
    const { volumeInfo, saleInfo, id } = d;
    const {
      imageLinks,
      title,
      subtitle,
      authors,
      publishedDate,
      averageRating,
      description,
      industryIdentifiers
    } = volumeInfo;
    const { listPrice } = saleInfo;
    const isbn =
      industryIdentifiers &&
      industryIdentifiers.length &&
      industryIdentifiers[0];
    result.push({
      id,
      volumeInfo: {
        imageLinks,
        title,
        subtitle,
        authors,
        publishedDate,
        averageRating
      },
      saleInfo: { listPrice },
      description,
      isbn
    });
  });
  return result;
};

export const flattenArray = style => {
  let result = {};
  style.forEach(element => {
    result = { ...result, ...element };
  });
  return result;
};

export const delay = duration =>
  new Promise(resolve => setTimeout(resolve, duration));

export const calculateBasketItemCount = basket => {
  return basket ? basket.reduce((acc, item) => acc + item.count, 0) : 0;
};
