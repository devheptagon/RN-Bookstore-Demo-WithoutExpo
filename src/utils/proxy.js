import api from "axios";

const Proxy = {
  sendRequest: function(httpVerb, url, data) {
    return api({
      url,
      headers: {
        "content-type": "application/json"
      },
      method: httpVerb,
      responseType: "json",
      data
    })
      .then(response => response.data)
      .catch(error => {
        console.error(error);
        alert(error);
      });
  },
  get: function(path) {
    return this.sendRequest("GET", path);
  },
  post: function(path, data) {
    return this.sendRequest("POST", path, data);
  },
  patch: function(path, data) {
    return this.sendRequest("PATCH", path, data);
  },
  delete: function(path, data) {
    return this.sendRequest("DELETE", path, data);
  }
};

export default Proxy;
