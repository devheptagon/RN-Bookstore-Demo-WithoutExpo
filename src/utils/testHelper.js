import React from 'react'
import { createStackNavigator, createAppContainer } from "react-navigation";
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'


export const wrapWithMockReduxAndNavigator = (Component) => {
  const initialState = { appReducer: {}}
  const mockStore = configureStore()

  const AppNavigator = createStackNavigator({
    Component
  })
  const ComponentWithNavigation = createAppContainer(AppNavigator)
  return <Provider store={mockStore(initialState)}>
      <ComponentWithNavigation />
    </Provider>   
}