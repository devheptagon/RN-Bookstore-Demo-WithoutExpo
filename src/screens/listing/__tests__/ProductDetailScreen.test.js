import React from "react";
import { ProductDetailScreen } from "../ProductDetailScreen";
import renderer from "react-test-renderer";
import { wrapWithMockReduxAndNavigator } from "../../../utils/testHelper";

test("ProductDetailScreen renders correctly", () => {
  const component = wrapWithMockReduxAndNavigator(ProductDetailScreen);
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
