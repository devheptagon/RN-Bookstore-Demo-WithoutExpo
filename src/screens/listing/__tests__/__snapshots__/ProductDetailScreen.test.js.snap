// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailScreen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    enabled={false}
    hitSlop={
      Object {
        "right": 50,
      }
    }
    maxDeltaY={20}
    minOffsetX={5}
    onGestureEvent={[Function]}
    onHandlerStateChange={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#eee",
          "flex": 1,
          "flexDirection": "column-reverse",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            style={
              Object {
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "padding": 5,
                    }
                  }
                >
                  <Image
                    source={1}
                    style={
                      Array [
                        Object {
                          "borderColor": "#eee",
                          "borderWidth": 2,
                        },
                        Object {
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "transform": Array [
                Object {
                  "translateX": -750,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderBottomColor": "#A7A7AA",
                "borderBottomWidth": 0.5,
                "height": 64,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
